pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_object.go

```diff
@@ -48,7 +48,20 @@ func init() {
 		func(from, to NodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.TypeFilter(rulesVersionKey, isDescriptor),
-				to.TypeFilter(rulesVersionKey, isSimpleDependent),
+				to.TypeFilter(rulesVersionKey, isSimpleDependent, Not(isConstraintDependent)),
+				JoinOnDescID(from, to, "desc-id"),
+				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_ABSENT),
+			}
+		})
+
+	registerDepRule(
```

Comment:
Can you comment that this is a weird rule that deals with the fact that these back-reference elements are bizarre? I think in practice we want to tie them to the name rather than to the descriptor to eliminate them.

Reply:
done